### DISCUSSION TITLE
Appending Filename with Model Names - Limitation w/Alternative

### DISCUSSION THREAD

**Discussion by jfreiman** [CodyBerenson/PGMA-Modernized#159] [General]
PGMA-M has added the ability to add the Model Names in a scene or feature by appending the filename with the names of the performers after the YEAR.  This works well for titles with only a couple of performers, but since filenames (at least in Windows) prevent adding all performers, for say an orgy or feature/compilation, this feature is not as useful as it could be.

The alternative way to add performer data (and my much-preferred method) is to add the performers to the MP4 or MKV metadata using Foobar 2000.

Just add the performers names to the ARTIST field separated by commas.

I have so far not discovered a limitation of how long this list can be and so far, Plex has had zero issues pulling that field into the service.

note: the performers photos aren't added to Plex via this method - only blank placeholders for each performer.

**Comment by JPH71** [comment]
Just seen this.....

The ability to add actor names to tiles was to aid in actor matching on
many scene blogs where links in the blog entry pointed to other metadata
other than cast members... This happened a lot with Queerclick OR where
tags in the blog did not only deal with actors, as in fagalicious...

The use of an actor list is optional especially for searchable movies as
the entry on a webpage usually has the cast list in full....

From memory Windows allows upto 127 Characters in a filename....

I will try and find a way for Plex to look at the Artists Metadata field
and see if one can pull their pictures...

Jason

**Reply by jfreiman** [reply]
Seeing if you can use the videos metadata to match and pull from AIFD would be incredible!
Thanks for considering it.

**Comment by JPH71** [comment]
This is done already... find the video on IAFD and name your download
accordingly...

Unless you mean summat else...

**Comment by JPH71** [comment]
Take a look at this!

https://github.com/squaresmile/Plex-Plug-Ins/blob/master/LocalMedia.bundle/Contents/Code/videohelpers.py#L24

**Reply by jfreiman** [reply]
> Take a look at this! https://github.com/squaresmile/Plex-Plug-Ins/blob/master/LocalMedia.bundle/Contents/Code/videohelpers.py#L24
I'm not quite sure what I'm supposed to be looking at - but I do see a few items labeled as 'new' under actor.....  
But, I'm excited if you're excited!!! I and my collections can't wait to test out your improved code!

**Comment by JPH71** [comment]
This isn't my code - but this agent seems to get information from mp4
tags...
add the agent and see if it grabs the cast list from your video files...
I will try and implement  this in the future .......

so that our agents can grab embedded metadata....

**Comment by jfreiman** [comment]
FYI, the existing Agent already grabs the metadata from MP4 files and adds it into the Plex video data -- from what I know, it always has collected the following data (at least for about 5 years since I switched over to Plex).

**Plex Native MP4 Metadata retrieval**

- Models (ARTIST) or (@ACTORS) [name1, name2, etc)
- Studio (ALBUM ARTIST)
- Distributor (COPYRIGHT)
- Genre (GENRE) [genre1/genre2/ etc)
- Collection(s) (ALBUM) (collection1/ collection2/ etc)
- Description (LDES)

What the existing agent does not do is match the Actors (photo data) to any of the PGMA databases - GEVI, IAFD, etc.

I have not experienced any limitations for the number of any of the above items when it comes to Plex Server gathering the data.  However, only the fist 2? Collections are shown below the Video and Actor photos. Otherwise, all Categories are available in Categories for easy searching.

The information in the video's metadata overrides data gathered from PGMA.

(forgive me if this was already known)

PS. I add the above metadata to all my files for easy sorting, searching, renaming files, and moving files via Foobar2000 (and use MP4 Video & Audio Tag Editor to add the LDES (Long Description) when PGMA can't match or has an inferior Description).

**Comment by JPH71** [comment]
Yes, that's why I said I would have to incorporate what's in this agent to
our agents by grabbing this metadata and running it against iafd...
I am also tempted to update the tags within the video files with the
scraped data... it will only work for mp4 files
I will talk to Cody about this...

**Comment by CodyBerenson** [comment]
Good afternoon.  My two cents:  dealing with MP4 tags is outside the scope of the scraping agents on this site. Updating the MP4 tags with our agents after a match is 100% a *no*.   Using tags to match and scrape from an indexing site is counter intuitive to how Plex is designed.  If someone wants to populate Plex metadata via MP4 tags, its certainly their prerogative and Plex offers built-in-functionality.  As is if they wish to create NFOs, or update metadata manually in Plex, or use SQLite Inserts and Updates directly in the Plex database on the backend, or a combination of any/all.  To each their own.  Re: the agents on this site, let's continue to use the Plex's platform-driven approach to matching using a standardized file naming convention.   Cheers.

**Reply by jfreiman** [reply]
> Good afternoon. My two cents: dealing with MP4 tags is outside the scope of the scraping agents on this site. Updating the MP4 tags with our agents after a match is 100% a _no_. Using tags to match and scrape from an indexing site is counter intuitive to how Plex is designed. If someone wants to populate Plex metadata via MP4 tags, its certainly their prerogative and Plex offers built-in-functionality. As is if they wish to create NFOs, or update metadata manually in Plex, or use SQLite Inserts and Updates directly in the Plex database on the backend, or a combination of any/all. To each their own. Re: the agents on this site, let's continue to use the Plex's platform-driven approach to matching using a standardized file naming convention. Cheers.

This makes perfect sense. TY
